addons/pos_pine_labs/models/pine_labs_pos_request.py

```diff
@@ -0,0 +1,76 @@
+import json
+import logging
+import requests
+
+from odoo import _
+
+REQUEST_TIMEOUT = 10
+PINE_LABS_AUTO_CANCEL_DURATION_MIN = 10
+ALLOWED_ENDPOINTS = ['UploadBilledTransaction', 'GetCloudBasedTxnStatus', 'CancelTransactionForced']
+ALLOWED_PAYMENT_MODES_MAPPING = {
+    'all': 0,
+    'card': 1,
+    'upi': 10,
+}
+
+_logger = logging.getLogger(__name__)
+
+
+def call_pine_labs(payment_method: object, endpoint: str, payload: dict) -> dict:
+    """
+    Make a request to Pine Labs POS API.
+
+    :return The content of the response parsed from json
+    """
+    if endpoint not in ALLOWED_ENDPOINTS:
+        raise ValueError(f"Invalid endpoint: '{endpoint}'. Allowed endpoints are: {ALLOWED_ENDPOINTS}")
+    payload |= pine_labs_request_body(endpoint == 'UploadBilledTransaction', payment_method=payment_method)
+    pine_labs_url = _get_pine_labs_url(payment_method=payment_method)
+    url = pine_labs_url + endpoint
+    try:
+        response = requests.post(url, json=payload, timeout=REQUEST_TIMEOUT)
+        response.raise_for_status()
+        res_json = response.json()
+    except requests.exceptions.ConnectionError as error:
+        _logger.warning('Connection Error: %r with the given URL %r', error, url)
+        return {'errorMessage': error}
+    except requests.exceptions.HTTPError as error:
+        _logger.warning('HTTPError: %r', error)
+        return {'errorMessage': error}
+    except requests.exceptions.Timeout as error:
+        _logger.warning('Timeout: %r', error)
+        return {'errorMessage': error}
+    except json.decoder.JSONDecodeError as error:
+        _logger.warning('JSONDecodeError: %r', error)
+        return {'errorMessage': error}
+    return res_json
```

Comment:
```suggestion
```

Just return directly inside the `try` clause. I see that you don't capture the generic `Exception` which for me is fine. Let the other type of error propagate if we don't know to handle.